File: examples/ios/Examples/Encrypt.m
#import <XCTest/XCTest.h>
#import <Realm/Realm.h>
#import <Foundation/Foundation.h>

@interface EncryptObjc : XCTestCase

@end

@implementation EncryptObjc

- (void)testEncrypt {
    // :code-block-start: encrypt
    // Generate a random encryption key
    NSMutableData *key = [NSMutableData dataWithLength:64];
    (void)SecRandomCopyBytes(kSecRandomDefault, key.length, (uint8_t *)key.mutableBytes);

    // Open the encrypted Realm file
    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
    config.encryptionKey = key;
    
    // :hide-start:
    // This is not required for the example, but is required for the test.
    // Omit from the example.
    config.inMemoryIdentifier = @"myEncryptedInMemoryRealm";
    // :hide-end:
    NSError *error = nil;
    RLMRealm *realm = [RLMRealm realmWithConfiguration:config error:&error];
    if (!realm) {
        // If the encryption key is wrong, `error` will say that it's an invalid database
        NSLog(@"Error opening realm: %@", error);
    } else {
        // Use the realm as normal...
    }
    // :code-block-end:
}

@end

File: docs/source/examples/Encrypt/encrypt.swift
// Generate a random encryption key
var key = Data(count: 64)
_ = key.withUnsafeMutableBytes { bytes in
    SecRandomCopyBytes(kSecRandomDefault, 64, bytes)
}

// Open the encrypted Realm file
let config = Realm.Configuration(encryptionKey: key)

do {
    let realm = try Realm(configuration: config)
    // Use the Realm as normal
    let dogs = realm.objects(Dog.self).filter("name contains 'Fido'")
    // ...

} catch let error as NSError {
    // If the encryption key is wrong, `error` will say that it's an invalid database
    fatalError("Error opening realm: \(error)")
}
